app/src/main/java/org/dhis2/usescases/datasets/datasetDetail/DataSetDetailActivity.kt

```diff
@@ -0,0 +1,242 @@
+        if (requestCode == FilterManager.OU_TREE && resultCode == Activity.RESULT_OK) {
+            filtersAdapter.notifyDataSetChanged()
+            updateFilters(filterManager.totalFilters)
+        }
+        super.onActivityResult(requestCode, resultCode, data)
+    }
+
+    override fun showHideFilter() {
+        val transition = ChangeBounds()
+        transition.duration = 200
+        TransitionManager.beginDelayedTransition(binding.backdropLayout, transition)
+        backDropActive = !backDropActive
+        val initSet = ConstraintSet()
+        initSet.clone(binding.backdropLayout)
+
+        if (backDropActive) {
+            initSet.connect(
+                R.id.eventsLayout,
+                ConstraintSet.TOP,
+                R.id.filterLayout,
+                ConstraintSet.BOTTOM,
+                50
+            )
+        } else {
+            initSet.connect(
+                R.id.eventsLayout,
+                ConstraintSet.TOP,
+                R.id.backdropGuideTop,
+                ConstraintSet.BOTTOM,
+                0
+            )
+        }
+        initSet.applyTo(binding.backdropLayout)
+
+        binding.filterOpen.visibility = if (backDropActive) View.VISIBLE else View.GONE
+    }
+
+    override fun clearFilters() {
+        filtersAdapter.notifyDataSetChanged()
+    }
+
+    override fun updateFilters(totalFilters: Int) {
+        binding.totalFilters = totalFilters
+        binding.executePendingBindings()
+    }
+
+    override fun openOrgUnitTreeSelector() {
+        val ouTreeIntent = Intent(this, OUTreeActivity::class.java)
+        startActivityForResult(ouTreeIntent, FilterManager.OU_TREE)
+    }
+
+    override fun showPeriodRequest(periodRequest: FilterManager.PeriodRequest) {
+        if (periodRequest == FilterManager.PeriodRequest.FROM_TO) {
+            DateUtils.getInstance().showFromToSelector(this) {
+                filterManager.addPeriod(it)
+            }
+        } else {
+            DateUtils.getInstance().showPeriodDialog(
+                this, { datePeriods -> filterManager.addPeriod(datePeriods) },
+                true
+            )
+        }
+    }
+
+    override fun setCatOptionComboFilter(
+        categoryOptionCombos: Pair<CategoryCombo, List<CategoryOptionCombo>>
+    ) {
+        filtersAdapter.addCatOptCombFilter(categoryOptionCombos)
+    }
+
+    @SuppressLint("RestrictedApi")
+    override fun setWritePermission(canWrite: Boolean) {
+        binding.addDatasetButton.visibility = if (canWrite) View.VISIBLE else View.GONE
+    }
+
+    override fun startNewDataSet() {
+        binding.addDatasetButton.isEnabled = false
+        val bundle = Bundle()
+        bundle.putString(Constants.DATA_SET_UID, dataSetUid)
+        startActivity(DataSetInitialActivity::class.java, bundle, false, false, null)
+    }
+
+    override fun openDataSet(dataSet: DataSetDetailModel) {
+        val bundle = Bundle()
```

Comment:
Could use 'with' method or 'run' extension function to avoid use all times bundle word